example/advanced/server-io/advanced_server_io.cpp

```diff
@@ -0,0 +1,914 @@
+//
+// Copyright (c) 2016-2017 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+//------------------------------------------------------------------------------
+//
+// Example: Advanced server, one io_context per thread
+//
+//------------------------------------------------------------------------------
+
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/beast/websocket.hpp>
+#include <boost/beast/version.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/signal_set.hpp>
+#include <boost/asio/steady_timer.hpp>
+#include <boost/make_unique.hpp>
+#include <boost/bind.hpp>
+#include <boost/config.hpp>
+#include <algorithm>
+#include <cstdlib>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <thread>
+#include <vector>
+
+using tcp = boost::asio::ip::tcp;               // from <boost/asio/ip/tcp.hpp>
+namespace http = boost::beast::http;            // from <boost/beast/http.hpp>
+namespace websocket = boost::beast::websocket;  // from <boost/beast/websocket.hpp>
+
+class round_robin
+{
+    using io_ptr = std::shared_ptr<boost::asio::io_context>;
+    using io_work = boost::asio::executor_work_guard<
+        boost::asio::io_context::executor_type>;
+
+    struct ctx
+    {
+        ctx(io_ptr io, io_work work)
+            : io {io}
+            , work {work}
+        {
+        }
+
+        ctx(ctx const&) = delete;
+        ctx& operator=(ctx const&) = delete;
+        ctx(ctx&&) = default;
+        ctx& operator=(ctx&&) = default;
+        ~ctx() = default;
+
+        io_ptr io;
+        io_work work;
+    }; // struct ctx
+
+    std::vector<ctx> ctx_;
+    std::size_t next_ {0};
+
+public:
+
+    round_robin(std::size_t size = 1)
+    {
+        size = std::max<std::size_t>(1, size);
+
+        for (std::size_t i = 0; i < size; ++i)
+        {
+            io_ptr io {std::make_shared<boost::asio::io_context>()};
+            ctx_.emplace_back(io, boost::asio::make_work_guard(*io));
+        }
+    }
+
+    round_robin(round_robin const&) = delete;
+    round_robin& operator=(round_robin const&) = delete;
+    round_robin(round_robin&&) = default;
+    round_robin& operator=(round_robin&&) = default;
+    ~round_robin() = default;
+
+    void
+    run()
+    {
+        std::vector<std::thread> threads;
+        for (auto& e : ctx_)
+        {
+            threads.emplace_back(
+                boost::bind(&boost::asio::io_context::run, e.io));
```

Comment:
I think we can use `std::bind`. Better yet, this should probably be a lambda since it will be more clear to readers.

Reply:
In general you should avoid taking the address of member functions of classes you don't own. (Especially if they're in they're standard/"semistandard"(

Reply:
I'll change it to use a lambda. Noted on avoiding taking the address of member functions of non-owned objects.